arctic/store/version_store.py

```diff
@@ -591,27 +587,57 @@ def _write_metadata_history(self, symbol, metadata, start_time, end_time=None, *
         VersionedItem containing .metadata
         """
         self._arctic_lib.check_quota()
-        version = {'_id': bson.ObjectId()}
-        version['symbol'] = symbol
-        version['metadata'] = metadata
-        version['_start_time'] = start_time
+        document = {'_id': bson.ObjectId()}
+        document['symbol'] = symbol
+        document['metadata'] = metadata
+        document['start_time'] = start_time
         if end_time is not None:
-            version['_end_time'] = end_time
+            document['end_time'] = end_time
 
-        handler = self._write_handler(version, symbol, metadata, **kwargs)
-        mongo_retry(handler.write)(self._arctic_lib, version, symbol, None, None, **kwargs)
+        handler = self._write_handler(document, symbol, metadata, **kwargs)
+        mongo_retry(handler.write)(self._arctic_lib, document, symbol, None, None, **kwargs)
 
-        # Insert the new version into the metadata DB
-        mongo_retry(self._metadata.insert_one)(version)
+        # Insert the new document into the metadata DB
+        mongo_retry(self._metadata.insert_one)(document)
 
         logger.debug('Finished writing metadata for %s', symbol)
 
-        self._publish_change(symbol, version)
-
         return VersionedItem(symbol=symbol, library=self._arctic_lib.get_name(), version=0, data=None, metadata=metadata)
 
+    def write_metadata_history(self, symbol, entries, times, **kwargs):
+        """
+        Manually write metadata entries in ._metadata collection
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        entries : `list of dicts`
+            list of metadata to be written
+        times : `list of datetime.datetimes`
+            time stamps of metadata
+        kwargs :
+            passed through to the write handler
+        """
+        if not (isinstance(entries, list) and isinstance(times, list)):
```

Comment:
suggestion: accept any list-like object and create a local copy before you append, you do not want to modify the input here